docs/howto/operate/add-aws-hub.md

```diff
@@ -0,0 +1,222 @@
+these environments vars for kops to use, so we do it manually).
+
+```bash
+export AWS_PROFILE=<cluster_name>
+export AWS_ACCESS_KEY_ID=$(aws configure get aws_access_key_id)
+export AWS_SECRET_ACCESS_KEY=$(aws configure get aws_secret_access_key)
+```
+
+## Create an AWS kops-based cluster
+
+1. From the root directory on this repo, `cd` into the `kops` directory
+2. Set up a *state* bucket for kops. This bucket will store the cluster
+[state](https://kops.sigs.k8s.io/state/).
+
+``` bash
+export KOPS_STATE_STORE=s3://2i2c-<hub-name>-kops-state
+aws s3 mb $KOPS_STATE_STORE --region <region>
+```
+
+```{note}
+Kops [recommends versioning the S3 bucket](https://kops.sigs.k8s.io/getting_started/aws/#cluster-state-storage) in case you ever need to revert or recover a previous state
+store.
+```
+
+### Create and render a kops config file
+
+You can use 
+of the existing [jsonnet](https://jsonnet.org/) specifications as a "template" for your cluster (for example, [here is the Farallon Institute specification](https://github.com/2i2c-org/pilot-hubs/blob/master/kops/farallon.jsonnet)).
+You may need to tweak zones, names and instances, the rest is boilerplate to create a
+kops-based cluster accordingly to some specification already outlined in 
+[#28](https://github.com/2i2c-org/pangeo-hubs/issues/28). Once you have your jsonnet
+specification ready, you need to render it to create the config file kops understand.
+
+1. Render the `kops` config file with
+
+```bash
+jsonnet <cluster_name>.jsonnet -y > <cluster_name>.kops.yaml
+```
+
+2. Regrettably, the rendering creates yaml file with with 3 dots at the end, you can 
+delete it with
+
+```bash
+sed -i '' -e '$ d' <cluster_name>.kops.yaml
+```
+
+```{note}
+In Linux you will need to use instead: `sed -i '$ d' <cluster_name>.kops.yaml`
+```
+
+### Create the cluster
+
+1. Create the cluster configuration and push it to s3 with
+
+```bash
+kops create -f <cluster_name>.kops.yaml
+```
+
+2. You will need to a ssh key pair before actually creating the cluster
+
+```bash
+ssh-keygen -f ssh-key
+mv ssh-key <cluster_name>.key
+mv ssh-key.pub <cluster_name>.key.pub
+```
+
+3. Build the cluster with the following command (notice that you are passing the ssh public key you just 
+created in the previous step)
+
+```bash
+kops update cluster <cluster_name>hub.k8s.local --yes --ssh-public-key <cluster_name>.key.pub --admin
+```
+
+```{note}
+The `--admin` at the end will modify your `~/.kube/config` file to point to the new 
+cluster.
+```
+
+If everything went as expected, the cluster will be created after some minutes.
+
+```{note}
+If you try to validate the cluster, validation will not pass until this next section is done.
+```
+
+4. Relocate and encrypt the generated keys
+
+After creating the cluster, you will need to relocate the public and private keys with
+
+```bash
+mv <cluster_name>.key ssh-keys/<cluster_name>.key
+mv <cluster_name>.key.pub ssh-keys/<cluster_name>.key.pub
```

Comment:
What do you think about not checking in the public key? We can always regenerate it from the private key with `ssh-keygen -y -f <cluster-name>.key`

Reply:
I agree, I will remove that line and make a note about how to recover it.
Btw, we should probably remove the public keys we have already pushed into the repo.